homeassistant/components/wmspro/generic_entity.py

```diff
@@ -0,0 +1,47 @@
+"""Generic entity for the WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+from wmspro.destination import Destination
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+
+class WebControlProGenericEntity(Entity):
+    """Foundation of all WMS based entities."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, dest: Destination) -> None:
+        """Initialize the entity with destination channel."""
+        super().__init__()
+        self._dest = dest
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.unique_id)},
+            manufacturer=MANUFACTURER,
+            model=self._dest.animationType.name,
+            name=self._dest.name,
+            serial_number=self._dest.id,
+            suggested_area=self._dest.room.name,
+            via_device=(DOMAIN, self._dest.host),
+            configuration_url=f"http://{self._dest.host}/control",
+        )
+
+    async def async_update(self) -> None:
+        """Update the entity."""
+        await self._dest.refresh()
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"dest_{self._dest.id}"
```

Comment:
You don't need to prefix the unique_id with `dest_`

Reply:
The unique_id returned from the hub is an integer, so I would at least call `str` on it.

Reply:
Oh that's fine